drafts/arcalinea-tx-expiry/draft.rst

```diff
@@ -0,0 +1,76 @@
+**Title:** Transaction Expiry
+
+**Author:** Jay Graber
+
+**Status:** Active
+
+**Category:** Standards
+
+**Created:** 2018-01-09
+
+Abstract
+===========
+
+This is an Standards ZIP describing a new consensus rule to set an expiration time for transactions that are not mined to be removed from the mempool.
+
+Motivation
+===========
+
+Transactions that are too large or have insufficient fees are often not mined. This indeterminism is a source of confusion for users and wallets. Allowing transactions to set a block height or timestamp after which it expires from the mempool would provide certainty around how long a transaction has to confirm before it is rejected by the network and must be re-sent.
+
+Advantages include improving performance by removing transactions unlikely to be mined, and potentially simplifying bidirectional payment channels by reducing the need to store and compress revocations for past states, since transactions not committed to the chain could expire and become invalid after a period of time.
+
+Specification
+===============
+
+Transactions will have a new field, nExpiryTime, which will set the block height or UNIX timestamp after which transactions will be removed from the mempool if they have not been mined.
+
+The data type for nExpiryTime will be uint32_t, conforming to the structure of nLockTime. If used in combination with nLockTime, both nLockTime and nExpiryTime must be the same unit (either block height or timestamp).
+
+For the example below, the last block that the transaction below could possibly be included in is 3539. After that, it will be removed from the mempool.
+
+````
+"txid": "17561b98cc77cd5a984bb959203e073b5f33cf14cbce90eb32b95ae2c796723f",
+"version": 3,
+"locktime": 2089,
+"blockexpiry": 3539,
+```
```

Comment:
Should we include what the block height was at the time the transaction was made? Maybe it's not necessary, but thought I would ask.

Reply:
I don't think it's necessary, is there a definite use case for this?